lib/Doctrine/MongoDB/Cursor.php

```diff
@@ -665,6 +669,22 @@ public function timeout($ms)
     }
 
     /**
+     * Wrapper method for MongoCursor::maxTimeMS().
+     *
+     * @see http://php.net/manual/en/mongocursor.maxtimems.php
+     * @param integer $ms
+     * @return self
+     */
+    public function maxTimeMS($ms)
+    {
+        if (version_compare(phpversion('mongo'), '1.5.0', '>=')) {
+            $this->maxTimeMS = (integer)$ms;
+            $this->mongoCursor->maxTimeMS((integer)$ms);
+        }
+        return $this;
```

Comment:
@ne0h12: The most compatible solution would be to call [`MongoCursor::addOption('$maxTimeMS', $ms)`](http://us2.php.net/manual/en/mongocursor.addoption.php) here.